flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/api/Table.java

```diff
@@ -1355,6 +1355,28 @@ default Table limit(int offset, int fetch) {
      */
     TableResult executeInsert(String tablePath, boolean overwrite);
 
+    /**
+     * Declares that the pipeline defined by the given {@link Table} object should be written to a
+     * table defined by a {@link TableDescriptor}. It executes the insert operation.
```

Comment:
side comment for the other `executeInsert` methods as well: explain that using the same descriptor will not lead to the same sink, we should recommend a statement set instead

Reply:
I added docs for now that note the behavior of registering the sink multiple times. I'm not sure I understand how a statement set solves this issue, though, at least using `StatementSet#addInsert(TableDescriptor, Table)` would have the same behavior, wouldn't it?

Ideally we'd probably avoid registering tables multiple times, and instead ensure that the same descriptor is only effectively registered once. But that'd require a solid/safe hash function that guarantees different values for different descriptors – or for the descriptor itself to generate a UUID which it exposes (yuck :-/)

Reply:
What I meant was a general description statement set vs. executeInsert. Statement set has the benefit of inserting into the same table from multiple pipelines in a single execution. This is not well explained in this method. I will add a paragraph as a hotfix commit while merging.